src/query/query.pest

```diff
@@ -10,17 +10,24 @@ expression         =  { expr }
 funcNames          = _{ latest }
 latest             =  { "latest" }
 
-infixExpression = { firstArg ~ infixFunction ~ secondArg }
+infixExpression = { firstArg ~ infixFunction ~ lookupValue }
 firstArg        = { lookup }
-secondArg       = { string }
-infixFunction   = { char* }
+infixFunction   = { char{1, 2} }
 
 lookup   = _{ lookupId | lookupName | lookupParam }
 lookupId = { "id" }
 lookupName = { "name" }
 lookupParam = { "parameter:" ~ lookupParamName }
-lookupParamName = { ASCII_ALPHANUMERIC* }
+lookupParamName = @{ (ASCII_ALPHANUMERIC | "_" )+ }
+lookupValue = { string | boolean | float | integer }
 
+float = @{
+             "-"?
+             ~ ("0" | ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*)
+             ~ ("." ~ ASCII_DIGIT*)
```

Comment:
not that we should , but numbers that this misses will include:

```
+5
1e-4
1e4
```

Reply:
Actually I don't know enough about how to read this to know how to parse it - does this exclude things with multiple dots in it? 

This stackoverflow answer looks like another approach? https://stackoverflow.com/a/68485567

Reply:
Yeah won't allow 1e4, deliberately left that out comment in the description. It wouldn't be too hard to add was just leaving it out for now to keep the already fairly confusing pattern as simple as possible. You can try it out here https://pest.rs/?g=N4Ig5gTghgtjURALhARwK4FMIE8AEBhhAvAcHgMoDyAkngH54BmANgPZQAuDeAorXgC%2BAHQB2Y1h26kAAsDFFFRYSAC0KgPwKlSxgAoVABhV4APngCCFAMI0aAfQByVRwC1eAJSr2AIjQDiNAAqPFa2Dn6BQQBUAJTaOkT6KgB0Joxhdr4BwXEJSoIgADQgAJaiAA7onMgghikAjABMAMwggkA#editor it excludes multiple dots. 

But I can add ability to write as +4, that should be fairly straightforward.

Reply:
Ok actually done this now, collapsed the numbers to all be floats instead of having the float int distinction and then this wasn't too painful to add